src/Symfony/Component/Intl/Tests/LanguagesTest.php

```diff
@@ -929,20 +1715,51 @@ public function testExists()
 
```

Comment:
In `testExists()` we should add this line:
```
  $this->assertFalse(Languages::exists('sh'));
```

Reply:
the language exists upstream: https://github.com/unicode-org/icu/blob/master/icu4c/source/data/lang/en.txt#L531

but im not sure it's an official iso code today: https://www.loc.gov/standards/iso639-2/php/langcodes_name.php?iso_639_1=sh

this may be politics related :) let's keep as-is for now (or propose a PR to exclude it)

Reply:
Then you should also make sure whe have a three letter code for it.

Reply:
there is none :thinking: https://github.com/unicode-org/icu/blob/2b3e3e3ce6f65bb15b84d837f857ecb93e176f7f/icu4c/source/data/misc/metadata.txt#L1234-L1237

ultimately this asks for "alias" support (to deal with legacy codes i guess), but im not sure we should provide those codes as valid actually.

Reply:
My point is that if we include `sh` we should also include a three letter alias for it. And if we decide it is lagacy, it should also be removed from the two letter list.

Reply:
yes, but that's a different topic IMHO. We implicitly obtain `sh` currently because it does hold a translation, we dont check the "legacy" at this point. So this is how things work today im afraid.

Reply:
This is the only reason the test would fail where you compare the values of `getAlpha3Names` with the values of `getNames`  (if you remove the localized longer that 3 letter entries).

Is that why you do not want to have that test?

I disagree and I think it is a bug that those lists do not contain the exact same values.

Reply:
if after #33148 and #33165 it is proved the lists are the same, we should test is yes :+1: